app/models/user.rb

```diff
@@ -1,4 +1,9 @@
 class User < ActiveRecord::Base
+  validates :email, uniqueness: { case_sensitive: false }
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
```

Comment:
Matt, reasonable remarks. Thanks! Let's discuss. 
I think we can exclude :database_authenticatable, :registerable, :recoverable,: trackable, :validatable, since we use OAuth. What about :rememberable? It manages generating and clearing token for remember the user from a saved cookie.